components/script/script_thread.rs

```diff
@@ -2393,6 +2403,19 @@ impl ScriptThread {
             }
         }
 
+        // Step 3 and 4
```

Comment:
This is only step 4

Reply:
`document.is_unloading` looks at `ignore_opens_during_unload_counter`, which is set by both `unload` and `prompt_to_unload`, so I was thinking this would in fact take care of both steps in one go. Perhaps I should rename the method to `is_unloading_or_prompting`?

Reply:
Looks like you're right!